src/adsbrain_server.cc

```diff
@@ -150,4 +178,426 @@ AdsBrainAPIServer::Handle(evhtp_request_t* req)
+      if (params_json.Find("binary_data_output", &bdo_json)) {
+        bool bdo = false;
+        RETURN_MSG_IF_ERR(
+            bdo_json.AsBool(&bdo), "Unable to parse 'binary_data_output'");
+        default_output_kind = (bdo) ? AllocPayload::OutputInfo::BINARY
+                                    : AllocPayload::OutputInfo::JSON;
+      }
+    }
+  }
+
+  // Get the byte-size for each input and from that get the blocks
+  // holding the data for that input
+  triton::common::TritonJson::Value inputs_json;
+  RETURN_MSG_IF_ERR(
+      request_json_ptr->MemberAsArray("inputs", &inputs_json),
+      "Unable to parse 'inputs'");
+
+  if (inputs_json.ArraySize() != 1) {
+    return TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INVALID_ARG,
+            (std::string("Only support one input now! Got  ") +
+            std::to_string(inputs_json.ArraySize())).c_str());
+  }
+
+  for (size_t i = 0; i < inputs_json.ArraySize(); i++) {
+    triton::common::TritonJson::Value request_input;
+    RETURN_IF_ERR(inputs_json.At(i, &request_input));
+    RETURN_IF_ERR(ValidateInputContentType(request_input));
+
+    const char* input_name;
+    size_t input_name_len;
+    RETURN_MSG_IF_ERR(
+        request_input.MemberAsString("name", &input_name, &input_name_len),
+        "Unable to parse 'name'");
+
+    const char* datatype;
+    size_t datatype_len;
+    RETURN_MSG_IF_ERR(
+        request_input.MemberAsString("datatype", &datatype, &datatype_len),
+        "Unable to parse 'datatype'");
+    const TRITONSERVER_DataType dtype = TRITONSERVER_StringToDataType(datatype);
+
+    triton::common::TritonJson::Value shape_json;
+    RETURN_MSG_IF_ERR(
+        request_input.MemberAsArray("shape", &shape_json),
+        "Unable to parse 'shape'");
+    std::vector<int64_t> shape_vec;
+    for (size_t i = 0; i < shape_json.ArraySize(); i++) {
+      uint64_t d = 0;
+      RETURN_MSG_IF_ERR(
+          shape_json.IndexAsUInt(i, &d), "Unable to parse 'shape'");
+      shape_vec.push_back(d);
+    }
+
+    RETURN_IF_ERR(TRITONSERVER_InferenceRequestAddInput(
+        irequest, input_name, dtype, &shape_vec[0], shape_vec.size()));
+
+    bool binary_input;
+    size_t byte_size;
+    RETURN_IF_ERR(
+        CheckBinaryInputData(request_input, &binary_input, &byte_size));
+
+    if ((byte_size == 0) && binary_input) {
+      RETURN_IF_ERR(TRITONSERVER_InferenceRequestAppendInputData(
+          irequest, input_name, nullptr, 0 /* byte_size */,
+          TRITONSERVER_MEMORY_CPU, 0 /* memory_type_id */));
+    } else if (binary_input) {
+      if (header_length == 0) {
+        return TRITONSERVER_ErrorNew(
+            TRITONSERVER_ERROR_INVALID_ARG,
+            "must specify valid 'Infer-Header-Content-Length' in request "
+            "header and 'binary_data_size' when passing inputs in binary "
+            "data format");
+      }
+
+      if (global_request_json_length_ > 0) {
+        uint32_t qlen = v[v_idx].iov_len;
```

Comment:
Add a comment about our assumption that: the size of request is smaller than one block (how big per block?)?